lib/ansible/plugins/action/nxos_file_copy.py

```diff
@@ -74,22 +75,22 @@ def process_playbook_values(self):
                 continue
             if argument_spec[key].get('type') is None:
                 argument_spec[key]['type'] = 'str'
-            type_ok = False
-            type = argument_spec[key]['type']
-            if type == 'str':
-                if isinstance(playvals[key], six.string_types):
-                    type_ok = True
-            elif type == 'int':
-                if isinstance(playvals[key], int):
-                    type_ok = True
-            elif type == 'bool':
-                if isinstance(playvals[key], bool):
-                    type_ok = True
-            else:
-                raise AnsibleError('Unrecognized type <{0}> for playbook parameter <{1}>'.format(type, key))
+            option_type = argument_spec[key]['type']
+            try:
+                if option_type == 'str':
+                    playvals[key] = validation.check_type_str(playvals[key])
+                elif option_type == 'int':
+                    playvals[key] = validation.check_type_int(playvals[key])
+                elif option_type == 'bool':
+                    playvals[key] = validation.check_type_bool(playvals[key])
+                elif option_type == 'path':
+                    playvals[key] = validation.check_type_path(playvals[key])
+                else:
+                    raise AnsibleError('Unrecognized type <{0}> for playbook parameter <{1}>'.format(type, key))
 
-            if not type_ok:
-                raise AnsibleError('Playbook parameter <{0}> value should be of type <{1}>'.format(key, type))
+            except (TypeError, ValueError) as e:
```

Comment:
@ganeshrn The tests in https://github.com/ansible/ansible/blob/devel/test/integration/targets/nxos_file_copy/tests/cli/input_validation.yaml need to be updated to handle the new error message.

Also, I tried running the test with this code bug I see the following error message:

```
The full traceback is:
Traceback (most recent call last):
  File "/root/agents-ci/ansible/lib/ansible/executor/task_executor.py", line 146, in run
    res = self._execute()
  File "/root/agents-ci/ansible/lib/ansible/executor/task_executor.py", line 645, in _execute
    result = self._handler.run(task_vars=variables)
  File "/root/agents-ci/ansible/lib/ansible/plugins/action/nxos_file_copy.py", line 499, in run
    self.file_push()
  File "/root/agents-ci/ansible/lib/ansible/plugins/action/nxos_file_copy.py", line 212, in file_push
    if not self.local_file_exists(local_file):
  File "/root/agents-ci/ansible/lib/ansible/plugins/action/nxos_file_copy.py", line 165, in local_file_exists
    return os.path.isfile(file)
  File "/usr/lib/python2.7/genericpath.py", line 37, in isfile
    st = os.stat(path)
TypeError: coercing to Unicode: need string or buffer, NoneType found
```

It does not appear to be generating the failure message `argument ... is of type..`

Reply:
Fixed review comments